client/utils/auth.py

```diff
@@ -140,5 +133,60 @@ def log_message(self, format, *args):
     update_storage(access_token, expires_in, refresh_token)
     return access_token
 
+
+def success_page(server, email):
+    """Generate HTML for the auth page - fetch courses and plug into templates"""
+    API = server + '/enrollment?email=%s' % email
+    data = urlopen(API).read().decode("utf-8")
+    return success_auth(success_courses(email, data, server))
+
+
+def success_courses(email, response, server):
+    """Generates HTML for individual courses"""
+    courses = json.loads(response)
+    if len(courses) > 0:
+        template_course = partial_course_html
+        html = ''
+        for course in courses:
+            html += template_course.format(**course)
+        status = 'Scroll for more: '+', '.join([c['display_name'] for c in courses])
+        byline = '"%s" is currently enrolled in %s.' % (email, pluralize(len(courses), ' course'))
+        title = 'Ok!'
+        head = ''
+    else:
+        html = partial_nocourse_html
+        byline = 'The email "%s" is not enrolled. Is it correct?' % email
+        status = 'No courses found'
+        title = 'Uh oh'
+        head = '<style>%s</style>' % red_css
+    return html, status, byline, title, head, server
+
+
+def success_auth(data):
+    """Generates finalized HTML"""
+    return auth_html.format(
+        site=data[5],
+        status=data[1],
+        courses=data[0],
+        byline=data[2],
+        title=data[3],
+        head=data[4])
+
+import os
+
+
+def get_file(relative_path, purpose):
+    dir = os.path.dirname(__file__)
+    filename = os.path.join(dir, relative_path)
+    return open(filename, purpose)
+
+
+def get_contents(relative_path, purpose='r'):
+    return get_file(relative_path, purpose).read()
+
+
+def pluralize(num, string):
```

Comment:
Not that it matters for this pull request, but this is not a very robust pluralizer, FYI... (e.g., watch -> watches)